apps/extension/src/features/auth/is-authed.tsx

```diff
@@ -34,3 +34,30 @@ export const useIsAuthed = () => {
 
   return { isAuthed, isAuthenticating };
 };
+
+/**
+ * Background job to validate if a trial is valid
+ */
+export const useValidateTrialOrSub = () => {
```

Comment:
Might refactor this and return the actual component later - but wanted to get the concept going first

Reply:
I don't understand the logic for the `component` useState.

`"children"` in router language means `<Outlet>` from what I understood. 

I wonder if instead of returning a component...why not move this logic to the top-level router?